libtransmission/utils.mm

```diff
@@ -0,0 +1,37 @@
+// This file Copyright © 2022-2023 Mnemosyne LLC.
+// It may be used under GPLv2 (SPDX: GPL-2.0-only), GPLv3 (SPDX: GPL-3.0-only),
+// or any future license endorsed by Mnemosyne LLC.
+// License text can be found in the licenses/ folder.
+
+#import <Foundation/Foundation.h>
+#include <string>
+#include <string_view>
+#include <utf8.h>
+
+// macOS implementation of tr_strv_replace_invalid() that autodetects the encoding.
+// This replaces the generic implementation of the function in utils.cc
+
+std::string tr_strv_replace_invalid(std::string_view sv, uint32_t replacement)
+{
+    // UTF-8 encoding
+    NSString* validUTF8 = @(std::string(sv).data());
```

Comment:
...but I'm not 100% sure we need a temporary std::string here? Seems like this would work too:

```suggestion
    NSString* validUTF8 = [NSString initWithBytes:std::data(sv) length:std::size(sv) encoding:[NSUTF8StringEncoding];
```

Reply:
I'll look at it after some sleep. :)